tests/integration/test_orders_api.py

```diff
@@ -536,6 +537,169 @@ async def test_download_asset_md(tmpdir, session):
     assert Path(filename).name == 'metadata.json'
 
 
+@respx.mock
+@pytest.mark.asyncio
+async def test_checksum_success(tmpdir, order_description, oid, session):
+    # Mock an HTTP response for download
+    order_description['state'] = 'success'
+    dl_url1 = TEST_DOWNLOAD_URL + '/asset1'
+    dl_url2 = TEST_DOWNLOAD_URL + '/asset2'
+    dl_url3 = TEST_DOWNLOAD_URL + '/manifest'
+    order_description['_links']['results'] = [{
+        'location': dl_url1
+    }, {
+        'location': dl_url2
+    }, {
+        'location': dl_url3
+    }]
+    get_url = f'{TEST_ORDERS_URL}/{oid}'
+    mock_asset = httpx.Response(HTTPStatus.OK, json=order_description)
+    respx.get(get_url).return_value = mock_asset
+
+    mock_asset1 = httpx.Response(HTTPStatus.OK,
+                                 json={
+                                     "md5": "mock_key1_md5",
+                                     "sha256": "mock_key1_sha256"
+                                 },
+                                 headers={
+                                     'Content-Type':
+                                     'application/json',
+                                     'Content-Disposition':
+                                     'attachment; filename="asset1.json"'
+                                 })
+    respx.get(dl_url1).return_value = mock_asset1
```

Comment:
Or... (an idea I had while running on Sunday) it could be helpful to simulate a TIFF asset instead of a JSON asset.

```suggestion
    # The first asset in the order is a 1 byte binary file containing the value "1".
    asset1_content = b"1"
    asset1_response = httpx.Response(HTTPStatus.OK,
                                 content=asset1_content, 
                                 headers={
                                     'Content-Type':
                                     'image/tiff',
                                     'Content-Disposition':
                                     'attachment; filename="asset1.tif"'
                                 })
    respx.get(dl_url1).return_value = asset1_response
```

Reply:
This makes sense, especially since the downloaded assets will include both json and TIFF files.